### ISSUE TITLE
Detect supported DRM systems

### ISSUE THREAD

**Issue by Dador** [google/shaka-player#3541]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes.

**Is your feature request related to a problem? Please describe.**
Related to #2786. Suggested solution is using `shaka.Player.probeSupport()`, but documentation for `shaka.Player.probeSupport()` says `This should only be used for diagnostics. NOTE: This may show a request to the user for permission`.
Sounds like `shaka.Player.probeSupport()` shouldn't be used in production.

**Describe the solution you'd like**
Maybe shaka can export `shaka.media.DrmEngine.probeSupport()` (currently not exported) or if `shaka.Player.probeSupport()` usage is acceptable is that case, then probably documentation should be changed.

My usecase: use HLS only if FairPlay is supported and use DASH in all other cases.

**Comment by joeyparrish** [comment]
You may just want to use `shaka.Player.probeSupport()` anyway.  The part of `probeSupport()` that may show a permissions prompt to the user is the part implemented in `DrmEngine`, because EME _may_ prompt the user for permission.  So if you want to use our probe system to detect supported key systems, you're going to risk a prompt either way.

Since you're interested in FairPlay specifically, there is an alternative.  You can query EME yourself, directly, and only query FairPlay, rather than the full list of key systems probed by `DrmEngine`.  FairPlay on Safari doesn't currently prompt the user as far as I know, so it may lower the risk of an undesirable UX if you do it that way.

I just tried this out in a Safari debugger, and it works:

```js
async function isFairPlaySupported() {
  config = {
    initDataTypes: ['cenc', 'sinf', 'skd'],
    videoCapabilities: [{contentType: 'video/mp4; codecs="avc1.42E01E"'}],
  };
  try {
    await navigator.requestMediaKeySystemAccess('com.apple.fps', [config]);
    return true;
  } catch (error) {
    return false;
  }
}

await isFairPlaySupported();
```

**Comment by Dador** [comment]
Thank you, it working perfectly!

Maybe it's good idea to add that snippet to documentation (for example on page [DRM Configuration](https://shaka-player-demo.appspot.com/docs/api/tutorial-drm-config.html) or page [FairPlay Support](https://shaka-player-demo.appspot.com/docs/api/tutorial-fairplay.html)).

I think it's popular use case: if you use DRM then probably you want to support Safari with Fairplay, so you need to detect FairPlay support in order to change source to HLS if FairPlay supported (AFAIK currently FairPlay is supported only within HLS and not publicly supported in DASH).

**Comment by joeyparrish** [comment]
Good idea.  I'll convert this to a "docs" issue, and we'll add this as a section in the FairPlay doc.

**Comment by avelad** [comment]
@joeyparrish With the commits linked to this issue, could it be closed?
